ntrans_gui.py

```diff
@@ -278,21 +282,53 @@ def generate_ntrans_dictionary(self) -> None:
                 return
 
         self.cancel_thread_event.clear()
+        self.generation_finished_event.clear()
 
         if self.thread is None:
             self.progress_queue: queue.Queue[int] = queue.Queue()
             glossary_generator = ntrans.GlossaryGenerator(
-                user_choices, self.progress_queue, self.cancel_thread_event
+                user_choices,
+                self.progress_queue,
+                self.cancel_thread_event,
+                self.generation_finished_event,
             )
             # Calls logic in ntrans.py
-            self.thread = threading.Thread(target=glossary_generator.read_ngram_files)
+            self.thread = threading.Thread(
+                target=glossary_generator.read_ngram_files, daemon=True
+            )
             self.thread.start()
 
             self.estimated_time_label.grid_remove()
             self.progress_frame.grid(
                 column=0, row=16, columnspan=2, padx=(0, 0), pady=(20, 40)
             )
+
             self.check_progressbar_queue()
+            self.check_if_finished()
+
+    def check_if_finished(self) -> None:
+        # Assert needed for mypy
+        assert self.thread is not None
+        if self.thread.is_alive():
+            self.root.after(500, self.check_if_finished)
+            return
+
+        if self.cancel_thread_event.is_set():
+            return
+
+        self.progress_frame.grid_remove()
+
+        finished_label1 = ttk.Label(self.on_finished_frame, text="Happy days!")
+        finished_label2 = ttk.Label(
+            self.on_finished_frame,
+            text="Your N-Trans Dictionary was successfully created!",
+        )
+        finished_label1.grid(column=0, columnspan=2)
+        finished_label2.grid(column=0, columnspan=2, padx=(5, 5))
+        self.on_finished_frame.grid(
+            column=0, row=16, columnspan=2, padx=(0, 0), pady=(20, 40)
+        )
+        self.thread = None
 
     def check_progressbar_queue(self) -> None:
```

Comment:
I don't think we need to separate `check_if_finished()` and `check_progressbar_queue()`. Currently, check_if_finished is basically this:

```
if thread is alive:
    schedule this to run again
    return
if cancel event set:
    return
show finished message
```

And `check_progressbar_queue` is like this:

```
update progressbar from queue
if thread is alive:
    schedule this to run again
else:
    hide progress bar
```

They could be combined into this:

```
if cancel event set:
    # Thread may or may not be alive, since it returns soon after canceling
    hide progress bar
elif thread is alive:
    # This is what happens most of the time
    update progressbar from queue
    schedule this to run again
else:
    # Thread died, but wasn't canceled
    # TODO: what if thread died because there was an error?
    show finished message
```